src/features/unified-share-modal/SharedLinkSection.js

```diff
@@ -51,11 +53,43 @@ type Props = {
     triggerCopyOnLoad?: boolean,
 };
 
-class SharedLinkSection extends React.Component<Props> {
+type State = {
+    isAutoCreatingSharedLink: boolean,
+};
+
+class SharedLinkSection extends React.Component<Props, State> {
     static defaultProps = {
         trackingProps: {},
+        autoCreateSharedLink: false,
     };
 
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            isAutoCreatingSharedLink: false,
+        };
+    }
+
+    componentDidUpdate(prevProps: Props) {
```

Comment:
```suggestion
    componentDidMount() {
        const { sharedLink, autoCreateSharedLink, addSharedLink, submitting } = this.props;

        if (
            autoCreateSharedLink &&
            !this.state.isAutoCreatingSharedLink &&
            !sharedLink.url &&
            !submitting &&
            !sharedLink.isNewSharedLink
        ) {
            this.setState({ isAutoCreatingSharedLink: true });
            addSharedLink();
        }
    }
    
    componentDidUpdate(prevProps: Props) { 
```

Reply:
If that doesn't work because of something like dummy data being passed in initially:

```suggestion
    // We handle didUpdate but not didMount because
    // the component initially renders with empty data
    // in order to start showing UI components.
    // When getInitialData completes in the parent we
    // rerender with correct sharedLink data and can
    // check whether to auto create a new one.
    componentDidUpdate(prevProps: Props) {
```

Reply:
This makes sense. my initial concern was that this could only cause the the shared link to be added before the rest of the USM was set up. the current USM code will trigger the DidUpdate version. I couldn't coerce it to do it on mount given the props that are being checked, but this might future-proof some refactoring.